docs: update `require-unicode-regexp.md` as following up #17402 (#17441)
eslint/eslint: docs/src/rules/require-unicode-regexp.md

## When Not To Use It

> If you don't want to notify regular expressions with no `u` and `v` flag, then it's safe to disable this rule.

Reviewer: ```suggestion
If you don't want to warn on regular expressions without either a `u` or a `v` flag, then it's safe to disable this rule.
```
